client/api/auth.js

```diff
@@ -30,7 +30,7 @@ function changePassword(currentPassword, newPassword) {
 }
 
 function getUserInfo() {
-  return request.get(urls.profile);
+  return request.base.get(urls.profile);
```

Comment:
What is the motivation behind this change? I assume it is to avoid the `auth` interceptor because of the fetch when the store is initiated. But I am wondering if there was a different approach since it seems to me to be fixing the symptom instead of the cause.

Reply:
The motivation behind introducing `client#base` (by @vladimyr I think) was to have a _pure_ Axios client with no interceptors attached to be used for auth calls like `login`. 
`getUserInfo` is primarily auth checking endpoint now that the only way to check auth and get user profile is using server so I see it as a good candidate for `client#base`.
I don't see that as a symptom per se, but intended behavior. I'm interested in how do you think this should be taken care of? Are you concerned that this is used for both profile fetch and auth check or that this is a slightly different API from other methods in this file (except `login`) that might not be so obvious?

Reply:
This then might be more of a philosophical question. If this is the only time we are calling this API endpoint and technically the user does not need to be authenticated to access it, should it be returning `FORBIDDEN` or `UNAUTHORIZED`? Should it return something else like `BAD_REQUEST` or even `OK` since we do not require them to be authenticated at that point. We just use the authentication flow to get the user if they are authenticated.

Reply:
Hmm, actually this is not the same case as `login` now that you mention it. Login is the one where authentication is not needed or to be more precise where `jwt` strategy is not used but authentication is done via basic strategy instead.
`/me` endpoint uses our _regular_ `jwt` strategy and does require authentication and I'm leaning on that fact in implementation. The only difference between `/login` and `/me` and other endpoints is that we have the handling in place for unauthorized errors and don't need an interceptor to play global authorization error handler. The issue is that interceptor comes before the error handling and not after so the only way to use it as a global error handler is to keep it as is and bypass it via `client#base` when not needed.
I wonder if it would be better to have a whitelist of routes where interceptor should not be triggered but I feel that would be a maintenance overhead and also I prefer this as it's more explicit and is usage driven (if `/me` is used on another place I don't want it to bypass interceptor).
@vladimyr @abasic Any thoughts, since I don't have a good and/or constructive idea here?

Reply:
I wanted to raise the question just to see if it was considered. Maybe a better question should have been, should using the JWT strategy always mean that the user is being authenticated. If it does then it should stay as is. If not then we could look into a different approach for such routes but that might be a general discussion and outside of this PR. I do not want to hold this up too much. Maybe we can talk tomorrow during the stand up about this topic.

Reply:
After an offline discussion, we decided to leave it as is. 
In an ideal world, checking authentication would be decoupled from fetching the profile (since those are semantically different) and the former would avoid interceptor and handle `UNAUTHORIZED` error as one of the expected responses while the latter would be treated as any other request.
Since in the current state of the codebase, there is no other use-case for profile fetch and the new strategy of storing the JWT requires at least one API call on the app boot to check for user auth this currently makes the most sense. (Since it at the same time checks for auth and hydrates client with user profile which is also needed on the app boot if the user is authorized, but was previously pulled from local storage).